device/src/main/java/com/softserve/lv460/device/action/ActionRegistry.java

```diff
@@ -0,0 +1,29 @@
+package com.softserve.lv460.device.action;
+
+import lombok.AllArgsConstructor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+@AllArgsConstructor
+public class ActionRegistry {
+  private DeviceAction deviceAction;
+
+
+  public List<Action> getAction(String type) {
+    return actionList().stream().filter((action) -> action.getType().equals(type))
+            .collect(Collectors.toList());
+  }
+
+
+  @Bean
+  List<Action> actionList() {
```

Comment:
you can define a ActionRegistry instance somewhere in @Configuration as a bean and inject all Action implementations:

@Bean
ActionRegistry actionRegistry(List<Action> actions) {
 return new ActionRegistry(List.of(new DeviceAction()));
}

Reply:
if you do like this - you should remove component annotation from both DeviceAction and ActionRegistry